src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -720,7 +720,14 @@ class SuggestAdapter {
 			return undefined;
 		}
 
-		const result: SuggestionDto = {
+		// 'overwrite[Before|After]'-logic
```

Comment:
I am not a fan of these changes - things are now spread out and much more complex (repeated conditions et al). I'm much more in favor of using some `!` expressions.

Reply:
I can revert this change if you feel strongly about this, but I disagree. The `!` suppression should only be used when required.  And by splitting the code into methods that handle converting one property, I find the code is more readable and that errors become obvious (this is how I caught the incorrect usage of `+=` instead of `|=` to create the CompletionItemInsertTextRule bitmask)

Reply:
Sorry, but feel strong about it. It would have not written the code like that and because I will continue to own it I wanna keep it close to what I am used to.

Reply:
Ok. I've reverted the extractions and added null suppressions + comments instead